workspace.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	tfe "github.com/hashicorp/go-tfe"
+)
+
+type WorkspaceConfig struct {
+	Terraform WorkspaceTerraform                      `json:"terraform"`
+	Variables map[string]WorkspaceVariable            `json:"variable,omitempty"`
+	Resources map[string]map[string]WorkspaceResource `json:"resource,omitempty"`
+}
+
+type WorkspaceBackend struct {
```

Comment:
Any suggestions on these type names? Figured I needed a prefix like Workspace since this is a shared namespace

Reply:
This seems fine for now, improving naming probably involves a new package